In [1]:
import pandas as pd

In [2]:
ca_data = pd.read_csv('../data/race_census/uscensusbureau-acs-2015-5-e-race/data/ca.csv')

In [3]:
ca_data.columns

Index(['summarylevel', 'state', 'statefips', 'countyfips', 'placefips',
       'cbsacode', 'congressdistrict', 'geoid', 'areaname', 'b02001_001',
       ...
       'c02003_010', 'c02003_011', 'c02003_012', 'c02003_013', 'c02003_014',
       'c02003_015', 'c02003_016', 'c02003_017', 'c02003_018', 'c02003_019'],
      dtype='object', length=225)

In [55]:
len(ca_data.columns)

225

We have 225 columns! We just need to read the metadata for this data explaining what each column means:

In [6]:
columns_metadata = pd.read_csv('../data/race_census/uscensusbureau-acs-2015-5-e-race/data/02_columnkey.csv')

In [8]:
for ci, column in columns_metadata.iterrows():
    print(column['columnid'], column['columndescription'])

B02001_001 RACE for Total Population
B02001_002 RACE for Total Population% White alone
B02001_003 RACE for Total Population% Black or African American alone
B02001_004 RACE for Total Population% American Indian and Alaska Native alone
B02001_005 RACE for Total Population% Asian alone
B02001_006 RACE for Total Population% Native Hawaiian and Other Pacific Islander alone
B02001_007 RACE for Total Population% Some other race alone
B02001_008 RACE for Total Population% Two or more races:
B02001_009 RACE for Total Population% Two or more races:% Two races including Some other race
B02001_010 RACE for Total Population% Two or more races:% Two races excluding Some other race, and three or more races
B02008_001 WHITE ALONE OR IN COMBINATION WITH ONE OR MORE OTHER RACES for White Alone Or In Combination With One Or More Other Races
B02009_001 BLACK OR AFRICAN AMERICAN ALONE OR IN COMBINATION WITH ONE OR MORE OTHER RACES for Black Or African American Alone Or In Combination With One Or More Othe

Looks like we are interested only in the first columns, from B02001_001 to B02001_008.

# Selecting only some columns

The code below select just the columns that we are interested and give proper names for the columns. The end result is the population number for all California counties.

In [46]:
# Select just some columns
select_columns = ['areaname'] + ['b02001_0{:02d}'.format(i) for i in range(1,9)]
ca_race = ca_data[select_columns]
# Use better names for the columns
ca_race.columns = ['areaname', 'total', 'white', 'black', 'native', 'asian', 'island', 'other', 'two']
ca_race

areaname     total     white  \
0                                            California  38421464  23747013   
1                                   California -- Urban  36538748  22201258   
2                                   California -- Rural   1882716   1545755   
3     California -- In metropolitan or micropolitan ...  38150399  23510834   
4        California -- In metropolitan statistical area  37587429  23038473   
5     California -- In metropolitan statistical area...  17529160  10041557   
6     California -- In metropolitan statistical area...  20058269  12996916   
7        California -- In micropolitan statistical area    562970    472361   
8     California -- In micropolitan statistical area...    158773    126352   
9     California -- In micropolitan statistical area...    404197    346009   
10    California -- Not in metropolitan or micropoli...    271065    236179   
11    California -- Not in metropolitan statistical ...    834035    708540   
12                           Alameda County, California   1584983    708558   
13                            Alpine County, California      1131       783   
14                            Amador County, California     36995     32006   
15                             Butte County, California    222564    185408   
16                         Calaveras County, California     44767     41210   
17                            Colusa County, California     21396     18188   
18                      Contra Costa County, California   1096068    663406   
19                         Del Norte County, California     27788     21248   
20                         El Dorado County, California    182093    157893   
21                            Fresno County, California    956749    570451   
22                             Glenn County, California     28029     23897   
23                          Humboldt County, California    135034    109821   
24                          Imperial County, California    178206    126305   
25                              Inyo County, California     18373     15200   
26                              Kern County, California    865736    646683   
27                             Kings County, California    150998    103289   
28                              Lake County, California     64158     51076   
29                            Lassen County, California     32645     24445   
...                                                 ...       ...       ...   
1649  Congressional District 24 (114th Congress), Ca...    722051    567426   
1650  Congressional District 25 (114th Congress), Ca...    718475    465375   
1651  Congressional District 26 (114th Congress), Ca...    717882    565051   
1652  Congressional District 27 (114th Congress), Ca...    716789    305717   
1653  Congressional District 28 (114th Congress), Ca...    716256    485646   
1654  Congressional District 29 (114th Congress), Ca...    714303    422407   
1655  Congressional District 30 (114th Congress), Ca...    745240    497241   
1656  Congressional District 31 (114th Congress), Ca...    729036    440201   
1657  Congressional District 32 (114th Congress), Ca...    718864    357861   
1658  Congressional District 33 (114th Congress), Ca...    715480    532001   
1659  Congressional District 34 (114th Congress), Ca...    714351    276306   
1660  Congressional District 35 (114th Congress), Ca...    727007    388200   
1661  Congressional District 36 (114th Congress), Ca...    733025    495313   
1662  Congressional District 37 (114th Congress), Ca...    719856    297831   
1663  Congressional District 38 (114th Congress), Ca...    717038    373711   
1664  Congressional District 39 (114th Congress), Ca...    722109    392463   
1665  Congressional District 40 (114th Congress), Ca...    709734    397807   
1666  Congressional District 41 (114th Congress), Ca...    729223    416327   
1667  Congressional District 42 (114th Congress), Ca...    749374    510254   
1668  Congressiona

# Data consistency

Verifying that the total is equal to the sum of the groups:

In [47]:
ca_race[['white', 'black', 'native', 'asian', 'island', 'other', 'two']].iloc[0].sum()

38421464

In [48]:
ca_race['total'].iloc[0]

38421464

# Now let's do it for all states

In [56]:
# The command belowe show that there is a file called usa_all_states.csv that countains 
# all the data already grouped by state

!ls ../data/race_census/uscensusbureau-acs-2015-5-e-race/data/

01_tablekey.csv   dc.csv  ks.csv  ms.csv  ny.csv  tn.csv	      wi.csv
02_columnkey.csv  de.csv  ky.csv  mt.csv  oh.csv  tx.csv	      wv.csv
ak.csv		  fl.csv  la.csv  nc.csv  ok.csv  usa_all_states.csv  wy.csv
al.csv		  ga.csv  ma.csv  nd.csv  or.csv  usa.csv
ar.csv		  hi.csv  md.csv  ne.csv  pa.csv  usa_zcta.csv
az.csv		  ia.csv  me.csv  nh.csv  pr.csv  ut.csv
ca.csv		  id.csv  mi.csv  nj.csv  ri.csv  va.csv
co.csv		  il.csv  mn.csv  nm.csv  sc.csv  vt.csv
ct.csv		  in.csv  mo.csv  nv.csv  sd.csv  wa.csv


In [57]:
usa = pd.read_csv('../data/race_census/uscensusbureau-acs-2015-5-e-race/data/usa_all_states.csv')
select_columns = ['state', 'areaname'] + ['b02001_0{:02d}'.format(i) for i in range(1,9)]
usa_selection = usa[select_columns]
usa_selection.columns = ['state', 'areaname', 'total', 'white', 'black', 'native', 'asian', 'island', 'other', 'two']

In [54]:
usa_selection.to_csv('../data/usa_race_states.csv')